iOS_SDK/OneSignalSDK/Source/OSStateSynchronizer.m

```diff
@@ -237,27 +229,23 @@ - (void)sendPurchases:(NSArray *)purchases appId:(NSString *)appId {
     if (!_currentSubscriptionState.userId)
         return;
     
-    let pushStateSyncronizer = [self getPushStateSynchronizer];
-    let emailStateSyncronizer = [self getEmailStateSynchronizer];
-    
+    let stateSyncronizer = [self getStateSynchronizers];
+    // Begin constructing the request for the external id update
     let requests = [NSMutableDictionary new];
-    requests[OS_PUSH] = [pushStateSyncronizer sendPurchases:purchases appId:appId userId:_currentSubscriptionState.userId externalIdAuthToken:_currentSubscriptionState.externalIdAuthCode];
-    
-    if (emailStateSyncronizer)
-        requests[OS_EMAIL] = [emailStateSyncronizer sendPurchases:purchases appId:appId userId:_currentEmailSubscriptionState.emailUserId externalIdAuthToken:_currentEmailSubscriptionState.emailAuthCode];
+    for (OSUserStateSynchronizer* userStateSynchronizer in stateSyncronizer) {
+        requests[userStateSynchronizer.getChannelId] = [userStateSynchronizer sendPurchases:purchases appId:appId];
+    }
 
     [OneSignalClient.sharedClient executeSimultaneousRequests:requests withSuccess:nil onFailure:nil];
 }
 
 - (void)sendBadgeCount:(NSNumber *)badgeCount appId:(NSString *)appId {
-    let pushStateSyncronizer = [self getPushStateSynchronizer];
-    let emailStateSyncronizer = [self getEmailStateSynchronizer];
-    
+    let stateSyncronizer = [self getStateSynchronizers];
+    // Begin constructing the request for the external id update
```

Comment:
This comment isn't correct, this is a bage update. Recommend removing instead of updating comment as the method name already describes the request about to be made.